### PR TITLE
Enable CPU on Parallel executor

### PR DESCRIPTION
**Pull request by chengduoZH**
Enable CPU on the Parallel executor.

### CODE REVIEW

**Comment by chengduoZH** [comment]
@luotao1 Certainly.

File: paddle/fluid/framework/details/CMakeLists.txt [hunk 1]
```diff
@@ -12,14 +12,16 @@ cc_library(ssa_graph_printer SRCS ssa_graph_printer.cc DEPS ssa_graph_builder)
 cc_library(variable_visitor SRCS variable_visitor.cc DEPS lod_tensor selected_rows)
 
 if(WITH_GPU)
-    nv_library(nccl_all_reduce_op_handle SRCS nccl_all_reduce_op_handle.cc DEPS op_handle_base scope lod_tensor ddim memory
+    nv_library(nccl_all_reduce_op_handle SRCS all_reduce_op_handle.cc DEPS op_handle_base scope lod_tensor ddim memory
```
Comment (reyoung) [inline paddle/fluid/framework/details/CMakeLists.txt]: `nccl_all_reduce_op_handle` --> `all_reduce_op_handle` ?

File: paddle/fluid/framework/details/all_reduce_op_handle.cc [hunk 2]
```diff
@@ -90,22 +100,25 @@ void NCCLAllReduceOpHandle::RunImpl() {
...
-                       ->Var()
+                       ->FindVar(out_var_handles[0]->name_)
                        ->GetMutable<framework::LoDTensor>();
 
       // Reduce All Tensor to trg in CPU
       ReduceLoDTensor func(lod_tensors, &trg);
       VisitDataType(ToDataType(lod_tensors[0]->type()), func);
 
-      for (size_t i = 0; i < local_scopes_.size(); ++i) {
+      for (size_t i = 1; i < local_scopes_.size(); ++i) {
```
Comment (reyoung) [inline paddle/fluid/framework/details/all_reduce_op_handle.cc]: This is not correct on GPU-CPU mixture code.

File: paddle/fluid/framework/details/CMakeLists.txt (hunk 1 above)
Comment (chengduoZH) [reply paddle/fluid/framework/details/CMakeLists.txt]: Done

File: paddle/fluid/framework/details/all_reduce_op_handle.cc (hunk 2 above)
Comment (chengduoZH) [reply paddle/fluid/framework/details/all_reduce_op_handle.cc]: Yes, you are right. 
But in the current implementation of `InsertNCCLAllReduceOp`, the inputs and outputs' `Variable` are the same respectively.
https://github.com/PaddlePaddle/Paddle/blob/1cfd3cb13b0ea4bf091757d59ec24e021563518f/paddle/fluid/framework/details/multi_devices_graph_builder.cc#L330-L340

So only to say, it doesn't cause error currently.

**Pull request merged**